fix: Address route scope DX comments (#524)
vaadin/docs: articles/flow/integrations/spring/scopes.asciidoc

You can inject other beans to the components, but note that while `@PostConstruct` methods work, `@PreDestroy` methods do not.
When you want to use another scope for the components, **you need to remember to add the `@SpringComponent` annotation to get the scope applied.**

.Changes in scopes need server restart
[NOTE]

> When changing scope annotations, the application server needs to be restarted to apply the new scopes, since the Spring IoC Container doesn't apply it on the fly even with enabled live reload.

Reviewer: The ordering of words is a bit funky, how about
```suggestion
When changing the scope annotations, the application server needs to be restarted to apply the new scopes. Restart is required even when live reload is enabled.
```

CAUTION: Injecting a "narrower" `RouteScope` bean into "wider" scope, like parent layout's `RouteScope` or `UIScope`, can cause problems.
For example, if you store a `RouteScope` bean into a `UIScope` bean, the bean might become stale after navigation.

> Usually `@RouteScopeOwner` qualifier has to be placed on top of the bean class and in the injection point of the bean.

Reviewer: ```suggestion
The `@RouteScopeOwner` qualifier has to be placed both on top of the bean class and on the injection point of the bean.
```

> In some certain cases, when a bean implementation can be resolved unambiguously by Spring IoC Container (as in the example above), `@RouteScopeOwner` annotation can be omitted in the injection point.

Reviewer: ```suggestion
But, when the bean implementation can be resolved unambiguously by Spring (as in the example above), the `@RouteScopeOwner` annotation can be omitted in the injection point.
```

However, it is recommended to have it there for better code readability.

Having an owner view class as a value in the `@RouteScopeOwner` on top of a model/business logic bean class ties application's view layer to a model/business layer.

> It can be decoupled by splitting the bean class into an interface and its implementation class, and then using the interface in the view class and marking a concrete bean's implementation class with `@RouteScopeOwner`.

Reviewer: ```suggestion
It can be decoupled for example by splitting the bean class into an interface and its implementation class, and then using the interface in the view class and marking a concrete bean's implementation class with `@RouteScopeOwner`.
```

> Having an owner view class as a value in the `@RouteScopeOwner` on top of a model/business logic bean class ties application's view layer to a model/business layer.
> It can be decoupled for example by splitting the bean class into an interface and its implementation class, and then using the interface in the view class and marking a concrete bean's implementation class with `@RouteScopeOwner`.

Reviewer 2: These are maybe a bit complex sentences.
```suggestion
Having an owner view class as a value in the `@RouteScopeOwner` for a model/business logic bean class ties application's view layer to a model/business layer.
It can be decoupled, for example, by splitting the bean class into an interface and its implementation class, and then using the interface in the view class and marking a concrete bean's implementation class with `@RouteScopeOwner`.
```

CAUTION: Injecting a "narrower" `RouteScope` bean into "wider" scope, like parent layout's `RouteScope` or `UIScope`, can cause problems.
For example, if you store a `RouteScope` bean into a `UIScope` bean, the bean might become stale after navigation.

> Usually `@RouteScopeOwner` qualifier has to be placed on top of bean class and injection point of the bean.

Author: ```suggestion
Usually `@RouteScopeOwner` qualifier has to be placed on top of the bean class and in the injection point of the bean.
```
